src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -75,7 +82,8 @@ uint64_t
 AsyncOpenTask::register_download_progress_notifier(std::function<SyncSession::ProgressNotifierCallback> callback)
 {
     if (auto session = m_session.load()) {
-        return session->register_progress_notifier(callback, SyncSession::ProgressDirection::download, false);
+        auto token = session->register_progress_notifier(callback, SyncSession::ProgressDirection::download, false);
```

Comment:
This token should be returned by the function

Reply:
🤦🏻‍♂️